runtime/frequency/src/lib.rs

```diff
@@ -1976,25 +2048,100 @@ sp_api::impl_runtime_apis! {
 	}
 
 	// Frequency runtime APIs
+	#[api_version(2)]
 	impl pallet_messages_runtime_api::MessagesRuntimeApi<Block> for Runtime {
 		fn get_messages_by_schema_and_block(schema_id: SchemaId, schema_payload_location: PayloadLocation, block_number: BlockNumber,) ->
 			Vec<MessageResponse> {
-			Messages::get_messages_by_schema_and_block(schema_id, schema_payload_location, block_number)
+			match Schemas::get_schema_by_id(schema_id) {
+				Some(SchemaResponseV2 { intent_id, .. }) => Messages::get_messages_by_intent_and_block(
+					intent_id,
+					schema_payload_location,
+					block_number,
+				).into_iter().map(|r| r.into()).collect(),
+				_ => vec![],
+			}
+		}
+
+		/// Retrieve the messages for a particular intent and block range (paginated)
+		fn get_messages_by_intent_id(intent_id: IntentId, pagination: BlockPaginationRequest) -> BlockPaginationResponse<MessageResponseV2> {
+			let mut response = BlockPaginationResponse::new();
```

Comment:
nit: This file is huge and I think it makes sense to keep these pallet specific details inside the pallet if possible